torch/csrc/jit/codegen/cuda/scheduler/utils.cpp

```diff
@@ -16,6 +13,120 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 namespace scheduler_utils {
+
+// Returns number of "valid" dimensions. e.g. if we have [I1, R2, I3, I4,
+// dontmerge[1]] result domain should be: [I1, I3*I4, R2, dontmerge[1]] with
+// return value 3
+// if we have [R1, I2, R3, I4, R4, dontmerge[0], dontmerge[1]] result domain
+// should be: [I2*I4, R1*R3, R4, dontmerge[0], dontmerge[1]] with return value 3
```

Comment:
Is `dontmerge[0]` indicating `R1` should not be merged? If so, why there is `R1*R3`?